internal/cache/persistent_lru_cache.go

```diff
@@ -231,6 +232,13 @@ func (c *PersistentCache) Put(ctx context.Context, key string, data gather.Bytes
 		// MUST NOT go over the specified limit for the cache space to avoid
 		// snapshots/restores from getting affected by the cache's storage use.
 		if c.isCacheFullLocked() {
+			// Limit warnings to one per minute max.
+			if clock.Now().Sub(c.lastCacheWarning) > time.Minute {
+				c.lastCacheWarning = clock.Now()
+
+				log(ctx).Warnf("Cache is full, unable to add %v into cache.", key)
```

Comment:
We should make this `Debug` down the road.

Reply:
This can be pretty noisy, even once per minute. Some workflows take several hours.

If the cache is full, are not we making the situation worse by filling up the logs ??

Reply:
@Shrekster yes, we should
- reduce the frequency
- add counters

Reply:
I discussed this partly offline. I think the write rate could be reduced, but even better I'd propose we just log once and then mute.

If we can settle on logging this once then this doesn't even need to be a debug log and that could be more useful upfront than having to turn on debug logs.

The alternate question to "how much logging is bad logging?" is maybe "what frequency is useful ?"

IMO once is ok for this becuase logging this the second time is only telling me that nothing got expired, but what debugging info do I get from it ? If cache was full once and the perf was crappy then that probably explains it.

Is the second log going to add more info ? Yes!

Does that more info help ? I don't know... doesn't seem like yet

Reply:
@Shrekster PTAL at #3193